cmd/transform/transform.go

```diff
@@ -0,0 +1,162 @@
+package transform
+
+import (
+	"context"
+	//"errors"
+	"os"
+	"path/filepath"
+
+	"github.com/konveyor/crane/internal/file"
+	"github.com/konveyor/crane/internal/plugin"
+	"github.com/konveyor/crane-lib/transform"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+type Options struct {
+	logger       logrus.FieldLogger
+	ExportDir    string
+	PluginDir    string
+	TransformDir string
+}
+
+func (o *Options) Complete(c *cobra.Command, args []string) error {
+	// TODO: @sseago
+	return nil
+}
+
+func (o *Options) Validate() error {
+	// TODO: @sseago
+	return nil
+}
+
+func (o *Options) Run() error {
+	return o.run()
+}
+
+func NewTransformCommand() *cobra.Command {
+	o := &Options{
+		logger: logrus.New(),
+	}
+	cmd := &cobra.Command{
+		Use:   "transform",
+		Short: "Create the transformations for the exported resources and plugins and save the results in a transform directory",
+		RunE: func(c *cobra.Command, args []string) error {
+			if err := o.Complete(c, args); err != nil {
+				return err
+			}
+			if err := o.Validate(); err != nil {
+				return err
+			}
+			if err := o.Run(); err != nil {
+				return err
+			}
+
+			return nil
+		},
+	}
+
+	addFlagsForOptions(o, cmd)
+
+	return cmd
+}
+
+func addFlagsForOptions(o *Options, cmd *cobra.Command) {
+	cmd.Flags().StringVarP(&o.ExportDir, "export-dir", "e", "export", "The path where the kubernetes resources are saved")
+	cmd.Flags().StringVarP(&o.PluginDir, "plugin-dir", "p", "plugins", "The path where binary plugins are located")
+	cmd.Flags().StringVarP(&o.TransformDir, "transform-dir", "t", "transform", "The path where files that contain the transformations are saved")
+}
+
+func (o *Options) run() error {
+	runner := transform.Runner{}
+
+	// Load all the resources from the export dir
+	exportDir, err := filepath.Abs(o.ExportDir)
+	if err != nil {
+		// Handle errors better for users.
+		return err
+	}
+
+	pluginDir, err := filepath.Abs(o.PluginDir)
```

Comment:
What if we don't want to run any plugins?

Reply:
If the plugin dir contains no plugins (or doesn't exist), then there will be no plugins processed. The result is that the transforms dir will contain all zero-length transforms (and there will be no whiteout files). Then again, if you intend to run no plugins, wouldn't simply not running transform be equivalent to this? Just skip transform and run apply. If apply doesn't find any whiteout or transform files, then the un-transformed resource files should end up as-is in the output dir, right?

Reply:
I was wondering if this should error at that point? 

I am also thinking about the future a little bit, but that is not fleshed out. I think this makes sense for now.

Reply:
Ahh. Yeah, we could eventually change this to error out here, but I'm thinking for now that if there are no plugins, just generating the empty patch files would be less disruptive on the cli workflows. But we can revisit as needed. For one thing, I have no idea how non-binary plugins will be configured once we add support there. The binary plugin dir may well be empty in cases where the plugins of interest are of other types.